MapboxCoreNavigation/RouteController.swift

```diff
@@ -640,12 +640,11 @@ extension RouteController: CLLocationManagerDelegate {
         }
         
         // Check and see if the user is near a future step.
-        guard let nearestStep = routeProgress.currentLegProgress.closestStep(to: location.coordinate),
-            nearestStep.index != routeProgress.currentLegProgress.stepIndex else {
+        guard let nearestStep = routeProgress.currentLegProgress.closestStep(to: location.coordinate) else {
             return false
         }
         
-        if nearestStep.distance < radius {
+        if nearestStep.distance < RouteControllerUserLocationSnappingDistance {
```

Comment:
This could also scale with radius as seen in https://github.com/mapbox/mapbox-navigation-ios/pull/1215/commits/6522939aa18041217e795491772e458bbecbbd63